test/e2e/sagemaker/resources/xgboost_hpojob.yaml

```diff
@@ -0,0 +1,64 @@
+apiVersion: sagemaker.services.k8s.aws/v1alpha1
+kind: HyperParameterTuningJob
+metadata:
+  name: $HPO_JOB_NAME
+spec:
+  hyperParameterTuningJobName: $HPO_JOB_NAME
+  tags:
+    - key: test-key
+      value: test-value
+  hyperParameterTuningJobConfig:
+    strategy: Bayesian
+    hyperParameterTuningJobObjective:
+      type_: Minimize
+      metricName: validation:error
+    resourceLimits:
+      maxNumberOfTrainingJobs: 10
+      maxParallelTrainingJobs: 5
+    parameterRanges:
+      integerParameterRanges:
+      - name: num_round
+        minValue: '10'
+        maxValue: '20'
+        scalingType: Linear
+      continuousParameterRanges: []
+      categoricalParameterRanges: []
+    trainingJobEarlyStoppingType: Auto
+  trainingJobDefinition:
+    staticHyperParameters:
+      base_score: '0.5'
+    algorithmSpecification:
+      trainingImage: $XGBOOST_IMAGE_URI
+      trainingInputMode: File
+    roleARN: $SAGEMAKER_EXECUTION_ROLE_ARN
+    inputDataConfig:
+    - channelName: train
+      dataSource:
+        s3DataSource:
+          s3DataType: S3Prefix
+          s3URI: s3://$SAGEMAKER_DATA_BUCKET/sagemaker/training/train
+          s3DataDistributionType: FullyReplicated
+      contentType: text/csv
+      compressionType: None
+      recordWrapperType: None
+      inputMode: File
+    - channelName: validation
+      dataSource:
+        s3DataSource:
+          s3DataType: S3Prefix
+          s3URI: s3://$SAGEMAKER_DATA_BUCKET/sagemaker/training/validation/
+          s3DataDistributionType: FullyReplicated
+      contentType: text/csv
+      compressionType: None
+      recordWrapperType: None
+      inputMode: File
+    outputDataConfig:
+      s3OutputPath: s3://$SAGEMAKER_DATA_BUCKET/sagemaker/training/hpo_output
```

Comment:
nit: Change to `/sagemaker/hpo/output`. I know HPO is technically running training, but it would help to differentiate the results of each integration test (if we choose to modify the HPO training job spec.